src/models/model.ts

```diff
@@ -36,7 +36,7 @@ export default class Model {
     }
   }
 
-  fromJSON(json?: any): this {
+  fromJSON(json: { [key: string]: any }): any {
```

Comment:
Previously we returned `this`, but changing it to `any` would break typing for chaining, would it not?

e.g. 
```js
model.fromJSON({ ... }).toJSON()
                         ^ typing would be broken and unable determine if `toJSON()` is valid
```

Reply:
@AaronDDM good catch! Not sure why I reverted it. In any case, it's back in now.